src/sieve.py

```diff
@@ -15,6 +15,11 @@ def sieve(n: int) -> list[int]:
     candidates = list(range(2, n + 1))
     primes = []
 
-    # FIXME: fill out this bit
-
+    while len(candidates) > 0:
+        p = candidates[0]
+        primes.append(p)
+        candidates.remove(p)
+        for number in candidates:
+            if (number/p)%1 == 0:
+                candidates.remove(number)
```

Comment:
Removing elements from a list is slow (unless it is the last) so it is a better solution to filter the entire list at once.

```python
while candidates:
    p = candidates[0]
    primes.append(p)
    candidates = [n for n in candidates if n%p == 0]
```